src/components/DraggableViewSection.tsx

```diff
@@ -33,54 +57,72 @@ function DraggableViewSection({ children, offset, dimension, vertical, dimension
     setHasLongHovered(false);
   }, []);
 
-  const handleDrag = useCallback((event: DragEvent<HTMLDivElement>) => {
-    if (!(event.clientX === 0 && event.clientY === 0)) {
-      if (prevDragPosition.current) {
-        const dimensionChange = vertical 
-          ? event.clientY - prevDragPosition.current.y
-          : event.clientX - prevDragPosition.current.x;
-        onDimensionChange(dimensionChange);
-      }
-      prevDragPosition.current = { x: event.clientX, y: event.clientY }
-    }
-  }, [onDimensionChange, vertical]);
+  const handleDrag = useCallback(
+    (_: DragEvent<HTMLDivElement | Body>) => {
+      setTimeout(() => {
+        if (dragEvent && !(dragEvent?.clientX === 0 && dragEvent?.clientY === 0)) {
+          if (prevDragPosition.current) {
+            const dimensionChange = vertical
+              ? (dragEvent.clientY ?? 0) - prevDragPosition.current.y
+              : (dragEvent.clientX ?? 0) - prevDragPosition.current.x;
+            onDimensionChange(dimensionChange);
+          }
+          prevDragPosition.current = {
+            x: dragEvent.clientX,
+            y: dragEvent.clientY,
+          };
+        }
+      });
+    },
+    [onDimensionChange, vertical]
+  );
 
   const handleDragStart = useCallback((event: DragEvent<HTMLDivElement>) => {
     setIsDragging(true);
     prevDragPosition.current = { x: event.clientX, y: event.clientY };
-  }, []);
+    document.body.addEventListener("dragover", dragoverHandler as any);
```

Comment:
I think it would be better to use `as unknown as EventListener` here so that it's clear to other developers what the intended type is